[pingcap/tidb#46609] support s3 profile for dumpling (#21456)
pingcap/docs: dumpling-overview.md

export AWS_SECRET_ACCESS_KEY=${SecretKey}
```

> Dumpling also supports reading credential files from `~/.aws/credentials` and authenticating with AWS profiles. For more information about URI parameter descriptions, see [URI Formats of External Storage Services](/external-storage-uri.md).

Reviewer: ```suggestion
Dumpling also supports reading credentials from `~/.aws/credentials` and authenticating with AWS profiles. For descriptions of URI parameters, see [URI Formats of External Storage Services](/external-storage-uri.md).
```

> To use the default credentials from `~/.aws/credentials`:

Reviewer: ```suggestion
To use credentials from the default AWS profile in `~/.aws/credentials`:
```

```shell
tiup dumpling -u root -P 4000 -h 127.0.0.1 -r 200000 -o "s3://${Bucket}/${Folder}"
```

> To use a specific AWS profile:

Reviewer: ```suggestion
To use credentials from a specific AWS profile:
```

export AWS_SECRET_ACCESS_KEY=${SecretKey}
```

Dumpling also supports reading credential files from `~/.aws/credentials` and using AWS profiles. For more information about URI parameter descriptions, see [URI Formats of External Storage Services](/external-storage-uri.md).

> 

Reviewer: for L239, do we need to provide two example commands so it is more clear for users? For instance:

- Use the default credentials from `~/.aws/credentials`:

```shell
tiup dumpling -u root -P 4000 -h 127.0.0.1 -r 200000 -o "s3://${Bucket}/${Folder}"
```

- Using a specific AWS profile:

```shell
export AWS_PROFILE=my-profile
tiup dumpling -u root -P 4000 -h 127.0.0.1 -r 200000 -o "s3://${Bucket}/${Folder}"
```

Author: thanks for suggestion. I will add it.

Author: I've added the examples.

export AWS_SECRET_ACCESS_KEY=${SecretKey}
```

Dumpling also supports reading credential files from `~/.aws/credentials` and authenticating with AWS profiles. For more information about URI parameter descriptions, see [URI Formats of External Storage Services](/external-storage-uri.md).

To use the default credentials from `~/.aws/credentials`:

```shell
tiup dumpling -u root -P 4000 -h 127.0.0.1 -r 200000 -o "s3://${Bucket}/${Folder}"
```

To use a specific AWS profile:

```shell

> export AWS_PROFILE=my-profile

Reviewer 2: as in the unit test of https://github.com/pingcap/tidb/pull/62199

I see we should specify profile as `s3://bucket/prefix/?profile=staging`, not the style of environment variables

Author: I'm really sorry, I fixed to use the option that  I added in the 
https://github.com/pingcap/tidb/pull/62199

Author: https://github.com/pingcap/docs/pull/21456/commits/870cc2c5e145221d8d552a81ced42b116374cf1d
